Cargo.toml

```diff
@@ -150,6 +150,7 @@ winreg = "0.10"
 # wasm
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
+gloo-timers = "0.2.6"
```

Comment:
I don't really know this dependency. I'm a little skeptical of taking on this, as opposed to just using js-sys or wasm-bindgen.

Reply:
I totally understand your concern. However, put my ocasional `rustwasm/gloo` contributer hat on, could you please reconsider this? There're some suprising papercuts with plain `wasm-bindgen` such as rustwasm/gloo#283 and using `gloo` gains better support in node & browsers & webworker, and most importantly, `gloo` is good maintaned under `rustwasm` organization, same as `wasm-bindgen`. cc @hamza1311  @futursolo as `rustwasm/gloo` maintainers.

I'm fine with your final decision, @seanmonstar

Reply:
It seems like the `Timeout` struct is just a wrapper for javascripts `setTimeout` which calls `clear_timeout` on drop. It might be a bit too much to add `gloo-timers` as a dependency.

Instead I would propose to create a `Timeout` struct in reqwest which calls `set_timeout_with_callback` from `web-sys` when created and calls `clear_timeout` on drop. This way we would get the best of both worlds (without having to import `gloo-timers`). Does that make sense @flisky? I can also work on this based on your PR btw, let me know!